src/dashboards/components/AutoRefreshInput.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {FC} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+
+// Components
+import DurationInput from 'src/shared/components/DurationInput'
+import {ButtonGroup} from '@influxdata/clockface'
+// Actions
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Types
+import {AutoRefreshStatus} from 'src/types'
+
+// Actions
+import {
+  setAutoRefreshInterval,
+  setAutoRefreshStatus,
+  resetDashboardAutoRefresh,
+  setAutoRefreshInputValue,
+} from 'src/shared/actions/autoRefresh'
+
+// Selectors
+import {getCurrentDashboardId} from 'src/dashboards/selectors'
+import {getAutoRefreshForDashboard} from 'src/shared/selectors/autoRefresh'
+
+const SUGGESTIONS = [
+  'None',
+  '10s',
+  '15s',
+  '30s',
+  '1m',
+  '5m',
+  '10m',
+  '30m',
+  '1h',
+]
+
+const AutoRefreshInput: FC = () => {
+  const dispatch = useDispatch()
+  const currentDashboardId = useSelector(getCurrentDashboardId)
+
+  const autoRefresh = useSelector(getAutoRefreshForDashboard)
+  const handleChooseAutoRefresh = (selection: string) => {
+    if (selection === 'None') {
+      dispatch(resetDashboardAutoRefresh(currentDashboardId))
+      return
+    }
+    const unit = selection[selection.length - 1]
+    let multiplier
+    if (unit === 's') {
+      multiplier = 1
+    } else if (unit === 'm') {
+      multiplier = 60
+    } else if (unit === 'h') {
+      multiplier = 3600
+    }
+
+    const calculatedMilliseconds =
+      Number(selection.slice(0, selection.length - 1)) * multiplier * 1000
+
+    if (unit === 's' || unit === 'm' || unit === 'h') {
+      dispatch(
+        setAutoRefreshInterval(currentDashboardId, calculatedMilliseconds)
+      )
+      dispatch(
+        setAutoRefreshStatus(currentDashboardId, AutoRefreshStatus.Active)
+      )
+      dispatch(setAutoRefreshInputValue(currentDashboardId, selection))
+    }
+  }
+
+  const isValidInput = (input: string) => {
+    const durationRegExp = /^(([0-9]+)(h|s|m))+$/g
+
+    return input === 'None' || !!input.match(durationRegExp)
+  }
+
+  return (
+    <ButtonGroup>
+      <DurationInput
+        submitInvalid={false}
+        value={autoRefresh?.refreshInputValue ?? 'None'}
+        onSubmit={handleChooseAutoRefresh}
+        suggestions={SUGGESTIONS}
+        customClass="refresh-input"
+        validFunction={isValidInput}
+        dividerText="Customize"
+        dividerOnClick={() => {
+          dispatch(
+            showOverlay('toggle-auto-refresh', null, () =>
+              dispatch(dismissOverlay())
+            )
```

Comment:
I don't have a good mental model right now of what's happening so sorry if I'm misunderstanding this, but it seems like at this point, the user has opened the overlay right? Why can't we just close it by dispatching the `dismissOverlay`?

Reply:
this is just the function passed through as the "onClose" handler, and follows a pattern we use elsewhere